cmd/restic/cmd_restore.go

```diff
@@ -141,7 +176,7 @@ func runRestore(ctx context.Context, opts RestoreOptions, gopts GlobalOptions, a
 		return err
 	}
 
-	res := restorer.NewRestorer(ctx, repo, sn, opts.Sparse)
+	res := restorer.NewRestorer(ctx, repo, sn, opts.Sparse, progressFormatter, term)
 
 	totalErrors := 0
 	res.Error = func(location string, err error) error {
```

Comment:
all calls to `Warnf` / `Verbosef` inside `runRestore` must use the `term` instead, otherwise we'll get a garbled output. (Take a look at `progressPrinter.P` and `progressPrinter.V` in cmd_backup.go)

Reply:
solved by overwriting `globalOptions.stdout` and `.stderr`. That way we can keep using `Warnf` and `Verbosef`.